torchtune/modules/moe/indices.py

```diff
@@ -0,0 +1,353 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
We should also add an acknowledgment to the original torchtitan code somewhere. @lessw2020 lmk if any preference on the best way to do this